packages/datadog-plugin-openai/src/index.js

```diff
@@ -1,1023 +1,17 @@
- */
-function normalizeRequestPayload (methodName, args) {
-  switch (methodName) {
-    case 'listModels':
-    case 'models.list':
-    case 'listFiles':
-    case 'files.list':
-    case 'listFineTunes':
-    case 'fine_tuning.jobs.list':
-    case 'fine-tune.list':
-      // no argument
-      return {}
-
-    case 'retrieveModel':
-    case 'models.retrieve':
-      return { id: args[0] }
-
-    case 'createFile':
-      return {
-        file: args[0],
-        purpose: args[1]
-      }
-
-    case 'deleteFile':
-    case 'files.del':
-    case 'retrieveFile':
-    case 'files.retrieve':
-    case 'downloadFile':
-    case 'files.retrieveContent':
-    case 'files.content':
-      return { file_id: args[0] }
-
-    case 'listFineTuneEvents':
-    case 'fine_tuning.jobs.listEvents':
-    case 'fine-tune.listEvents':
-      return {
-        fine_tune_id: args[0],
-        stream: args[1] // undocumented
-      }
-
-    case 'retrieveFineTune':
-    case 'fine_tuning.jobs.retrieve':
-    case 'fine-tune.retrieve':
-    case 'deleteModel':
-    case 'models.del':
-    case 'cancelFineTune':
-    case 'fine_tuning.jobs.cancel':
-    case 'fine-tune.cancel':
-      return { fine_tune_id: args[0] }
-
-    case 'createImageEdit':
-      return {
-        file: args[0],
-        prompt: args[1], // Note: order of prompt/mask in Node.js lib differs from public docs
-        mask: args[2],
-        n: args[3],
-        size: args[4],
-        response_format: args[5],
-        user: args[6]
-      }
-
-    case 'createImageVariation':
-      return {
-        file: args[0],
-        n: args[1],
-        size: args[2],
-        response_format: args[3],
-        user: args[4]
-      }
-
-    case 'createTranscription':
-    case 'createTranslation':
-      return {
-        file: args[0],
-        model: args[1],
-        prompt: args[2],
-        response_format: args[3],
-        temperature: args[4],
-        language: args[5] // only used for createTranscription
-      }
-  }
-
-  // Remaining OpenAI methods take a single object argument
-  return args[0]
-}
-
-/**
- * Converts an array of tokens to a string
- * If input is already a string it's returned
- * In either case the value is truncated
-
- * It's intentional that the array be truncated arbitrarily, e.g. "[999, 888, 77..."
-
- * "foo" -> "foo"
- * [1,2,3] -> "[1, 2, 3]"
- */
-function normalizeStringOrTokenArray (input, truncate) {
-  const normalized = Array.isArray(input)
-    ? `[${input.join(', ')}]` // "[1, 2, 999]"
-    : input // "foo"
-  return truncate ? truncateText(normalized) : normalized
-}
-
-function defensiveArrayLength (maybeArray) {
-  if (maybeArray) {
-    if (Array.isArray(maybeArray)) {
-      return maybeArray.length
-    } else {
-      // case of a singular item (ie body.training_file vs body.training_files)
-      return 1
-    }
-  }
-
-  return undefined
-}
-
-module.exports = OpenApiPlugin
```

Comment:
Does OpenAI still use Open API to build it's API?

Reply:
yeah they [do](https://github.com/openai/openai-node/blob/622c80aaa17c486cbd16cf620b3bee6b73650ba4/src/index.ts#L1) 😆